### DISCUSSION TITLE
Animated TabBar v3

### DISCUSSION THREAD

**Discussion by gorhom** [gorhom/react-native-animated-tabbar#118] [Announcements]
## Overview 

Since the release of `Reanimated v2` and I was busy with [BottomSheet](https://github.com/gorhom/react-native-bottom-sheet), and did not got enough time to upgrade this library.

The new release `v3` will need to address the following: 

- Split presets and core into separate packages `monorepo`.
- Rewrite code base to use `Reanimated v2`.
- Allow users to write their own custom tab bars.
- Add `React Navigation v6` adapter.
- Add accessibility support. 
- Add documentation website

The presets split is necessary to allow each preset to install its own per-dependencies  without worrying about other presets, also this will shrink the installed preset package size significantly. 

So if you want to install the `flashy` preset, you will install `@gorhom/animated-tabbar-flashy` package.

```
`@gorhom/animated-tabbar-flashy`
|-- `@gorhom/animated-tabbar-core` #dependency 
`-- `react-native-reanimated` # per-dependency
```

### Tasks
- [ ] #119 
  - [ ] extract `core functionality` into a separate package.
  - [ ] extract `flashy` preset into a separate package.
  - [ ] extract `material` preset into a separate package.
  - [ ] extract `bubble` preset into a separate package.
- [ ] #120.
- [ ] add `React Navigation v6` adapter.
- [ ] add accessibility support. 
- [ ] add documentation website.

**Comment by kamalmalaklife** [comment]
Almost 2 years after, any news on supporting navigation v6 ?
